### PR TITLE
Make infer_schema! skip views

### PR DESCRIPTION
**Pull request by killercup**
Rebased version of #374. Added some test migrations, will add actual tests later on.

### CODE REVIEW

**Comment by killercup** [comment]
@sgrif I'm not sure how to elegantly test the Postgres version. For SQLite we can just open a new `:memory:` database, create tables and views, and see that it works (i.e. ignore the views). For Postgres this is more tricky. Would it make sense to write compile-tests for this? Or do we just trust that it works based on the fact that we call `infer_schema!` in `diesel_test` and the migrations now include views?

**Comment by sgrif** [comment]
You should be able to just create the view in a transaction and it'll be fine.

**Comment by killercup** [comment]
Added a simple test for postgres. Tell me if you want to have more.

File: migrations/postgresql/20160909174741_create_highlight_post_view/up.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE stuff (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR NOT NULL,
+  important BOOLEAN NOT NULL DEFAULT 'f'
+);
+
+CREATE VIEW important_stuff AS
```
Comment (sgrif) [inline migrations/postgresql/20160909174741_create_highlight_post_view/up.sql]: Do you think we need this since we have the test on `load_table_names`?

Comment (killercup) [inline migrations/postgresql/20160909174741_create_highlight_post_view/up.sql]: Good question. I think it'd be fine to leave it in but currently it doesn't give us anything, so I'll remove it. If we were to support views later, we can just re-add these migrations.

**Comment by killercup** [comment]
Feel free to merge or squash-merge this as soon as travis likes it. I just pushed a revert for the migrations.

**Pull request merged**